lib/ruby_ui/separator/separator.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module RubyUI
+  class Separator < Base
+    ORIENTATIONS = %i[horizontal vertical].freeze
+
+    def initialize(as: "div", orientation: :horizontal, decorative: true, **attrs)
+      raise ArgumentError, "Invalid orientation: #{orientation}" unless ORIENTATIONS.include?(orientation.to_sym)
+
+      @as = as
+      @orientation = orientation.to_sym
+      @decorative = decorative
+      super(**attrs)
+    end
+
+    def view_template(&)
+      public_send(@as, **attrs, &)
```

Comment:
The `as` parameter could be a symbol, and then we could use `tag` phlex method here, eg.
```suggestion
      tag @as, **attrs, &
```

Reply:
I added it as `tag(@as, **attrs, &)` because, without the parentheses, the syntax highlighting was incorrect.

![image](https://github.com/user-attachments/assets/0125e93a-9316-4e6a-9dfa-a978180af337)